### PR TITLE
Add io.gpt4all.gpt4all

### PR DESCRIPTION
**Pull request by Tim453**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
Qt based GUI for GPT4All to run open source LLM's
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [ ] I am using only the minimal set of permissions. 
 It uses `xdg-documents:ro` to make use of gpt4all's local docs feature
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/nomic-ai/gpt4all/issues/698
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: io.gpt4all.gpt4all.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+    <id>io.gpt4all.gpt4all</id>
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>MIT</project_license>
+    <name>GPT4ALL</name>
+    <summary>Open-source assistant</summary>
+    <developer id="tld.vendor">
```
Comment (hfiguiere) [inline io.gpt4all.gpt4all.appdata.xml:8]: this developer is incorrect (copied from the example)

File: io.gpt4all.gpt4all.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        </screenshot>
+    </screenshots>
+    <url type="homepage">https://gpt4all.io</url>
+    <url type="bugtracker">https://github.com/nomic-ai/gpt4all/issues</url>
+    <url type="vcs-browser">https://github.com/nomic-ai/gpt4all</url>
+    <releases>
+      <release version="2.7.5" date="2024-05-03"></release>              
+    </releases>
+    <launchable type="desktop-id">io.gpt4all.gpt4all.desktop</launchable>
+    <content_rating type="oars-1.0">
```
Comment (hfiguiere) [inline io.gpt4all.gpt4all.appdata.xml:38]:
```suggestion
    <content_rating type="oars-1.1">
```

File: io.gpt4all.gpt4all.yml [hunk 3]
```diff
@@ -0,0 +1,136 @@
+app-id: io.gpt4all.gpt4all
+runtime: org.kde.Platform
+runtime-version: '6.6'
```
Comment (hfiguiere) [inline io.gpt4all.gpt4all.yml:3]: 6.7 is available.

File: io.gpt4all.gpt4all.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+base-version: '6.6'
+cleanup-commands:
+  - /app/cleanup-BaseApp.sh
+finish-args:
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=network
+  - --device=dri
+  - --env=LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/app/lib/x86_64-linux-gnu/
```
Comment (hfiguiere) [inline io.gpt4all.gpt4all.yml:15]: this won't work as variable substitutions aren't done. Also I'm sure it's unecessary. Also what does this do on `aarch64`?

File: io.gpt4all.gpt4all.appdata.xml
Comment (hfiguiere) [inline io.gpt4all.gpt4all.appdata.xml:1]: this file should be submitted upstream

Comment (hfiguiere) [reply io.gpt4all.gpt4all.appdata.xml:1]: why no using this: https://github.com/nomic-ai/gpt4all/blob/main/gpt4all-chat/flatpak-manifest/io.gpt4all.gpt4all.appdata.xml instead?

File: io.gpt4all.gpt4all.yml (hunk 4 above)
Comment (Tim453) [reply io.gpt4all.gpt4all.yml:15]: The problem is that the runtime linker does not find `libQt6HttpServer.so.6`, when it is not in the linker path

File: io.gpt4all.gpt4all.appdata.xml
Comment (Tim453) [reply io.gpt4all.gpt4all.appdata.xml:1]: You are right, I will submit the changes needed there.

File: io.gpt4all.gpt4all.yml (hunk 4 above)
Comment (hfiguiere) [reply io.gpt4all.gpt4all.yml:15]: if you mean "at build time" this is even more wrong.

Comment (Tim453) [reply io.gpt4all.gpt4all.yml:15]: Not at build time. When I try to start the application.

Comment (bbhtt) [reply io.gpt4all.gpt4all.yml:15]: It should be installed to `/app/lib`, if not you can add a symlink there.

**Comment by Tim453** [comment]
bot, build io.gpt4all.gpt4all

**Comment by Tim453** [comment]
bot, build io.gpt4all.gpt4all

File: io.gpt4all.gpt4all.yml [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+      - type: git
+        url: https://github.com/nomic-ai/gpt4all.git
+        tag: v2.8.0
+        commit: 09dd3dc31871118f9b3593e57b914016da0a0028
+
+  # aarch64 version without CUDA backend
+  # https://forums.developer.nvidia.com/t/nvcc-fails-to-build-with-arm-neon-instructions-cpp-vs-cu/248355
+  - name: gpt4all
+    only-arches:
+      - aarch64
```
Comment (bbhtt) [inline io.gpt4all.gpt4all.yml:167]: You can use per arch config-opts:

```yaml
    build-options:
      arch:
        x86_64:
          config-opts: 
```

**Comment by Tim453** [comment]
bot, build io.gpt4all.gpt4all

**Comment by Gamey404** [comment]
No clue if you need the feedback but it seems to work great on my system running Fedora 40! I installed it and loaded up llama 3 Instruct without any.

**Comment by ThiloteE** [comment]
I tried this pr on Nobara Linux 39. Got some signatures related error:

```
thilo@ThiloLinuxGamingAI:~$ flatpak install --user https://dl.flathub.org/build-repo/106306/io.gpt4all.gpt4all.flatpakref
Required runtime for io.gpt4all.gpt4all/x86_64/test (runtime/org.kde.Platform/x86_64/6.7) found in remote flathub
Möchten Sie es installieren? [Y/n]: y

io.gpt4all.gpt4all Berechtigungen:
    ipc     network     fallback-x11    wayland    x11    dri    file access [1]    dbus access [2]

    [1] xdg-config/kdeglobals:ro, xdg-documents:ro
    [2] com.canonical.AppMenu.Registrar, org.kde.KGlobalSettings, org.kde.kconfig.notify


        KENNUNG                                         Zweig                   Op             Gegenstelle                Herunterladen
 1. [✗] io.gpt4all.gpt4all.Locale                       test                    i              gpt4all-origin               1,0 KB / 61,0 MB
 2. [✓] org.freedesktop.Platform.GL.default             23.08                   u              flathub                      5,8 MB / 164,6 MB
 3. [✓] org.freedesktop.Platform.GL.default             23.08-extra             u              flathub                      3,8 MB / 164,6 MB
 4. [✓] org.kde.Platform.Locale                         6.7                     i              flathub                    319,9 KB / 380,7 MB
 5. [✓] org.kde.Platform                                6.7                     i              flathub                    164,6 MB / 326,9 MB
 6. [✗] io.gpt4all.gpt4all                              test                    i              gpt4all-origin             228,0 MB / 510,2 MB

Warning: Während des Holens von runtime/io.gpt4all.gpt4all.Locale/x86_64/test von der Gegenstelle gpt4all-origin: Commit 16dbe04e4f4623e06b40bf10dbbdd8bd67246bdc6e31753e4e592f844083f08a: GPG verification enabled, but no signatures found (use gpg-verify=false in remote config to disable)
Error: Während des Holens von app/io.gpt4all.gpt4all/x86_64/test von der Gegenstelle gpt4all-origin: Server returned HTTP 404
Fehler: Failed to install io.gpt4all.gpt4all: Während des Holens von app/io.gpt4all.gpt4all/x86_64/test von der Gegenstelle gpt4all-origin: Server returned HTTP 404
```

**Comment by rizitis** [comment]
```1] xdg-config/kdeglobals:ro, xdg-documents:ro
    [2] com.canonical.AppMenu.Registrar, org.kde.KGlobalSettings, org.kde.kconfig.notify


        ID                                                          Branch                   Op              Remote                      Download
 1. [✗] io.gpt4all.gpt4all.Locale                                   test                     i               gpt4all-origin                1.0 kB / 61.0 MB
 2. [✓] org.freedesktop.Platform.GL.default                         23.08                    i               flathub                     164.3 MB / 164.6 MB
 3. [✓] org.freedesktop.Platform.GL.default                         23.08-extra              i               flathub                      18.5 MB / 164.6 MB
 4. [✓] org.freedesktop.Platform.GL.nvidia-550-78                   1.4                      i               flathub                     307.7 MB / 307.8 MB
 5. [✓] org.freedesktop.Platform.VAAPI.Intel                        23.08                    i               flathub                      13.3 MB / 13.4 MB
 6. [✓] org.freedesktop.Platform.openh264                           2.2.0                    i               flathub                     886.7 kB / 944.3 kB
 7. [✓] org.kde.Platform.Locale                                     6.7                      i               flathub                      18.0 kB / 380.7 MB
 8. [✓] org.kde.WaylandDecoration.QAdwaitaDecorations               6.7                      i               flathub                       1.2 MB / 1.2 MB
 9. [✓] org.kde.Platform                                            6.7                      i               flathub                     259.4 MB / 326.9 MB
10. [✗] io.gpt4all.gpt4all                                          test                     i               gpt4all-origin              228.0 MB / 510.2 MB

Warning: While pulling runtime/io.gpt4all.gpt4all.Locale/x86_64/test from remote gpt4all-origin: Commit 16dbe04e4f4623e06b40bf10dbbdd8bd67246bdc6e31753e4e592f844083f08a: GPG verification enabled, but no signatures found (use gpg-verify=false in remote config to disable)
Error: While pulling app/io.gpt4all.gpt4all/x86_64/test from remote gpt4all-origin: While fetching https://dl.flathub.org/build-repo/106306/objects/fc/e978a2ccf30070652a18df44c2ee21b7db061ae9951388c1e26241f368c279.filez: Server returned HTTP 404
error: Failed to install io.gpt4all.gpt4all: While pulling app/io.gpt4all.gpt4all/x86_64/test from remote gpt4all-origin: While fetching https://dl.flathub.org/build-repo/106306/objects/fc/e978a2ccf30070652a18df44c2ee21b7db061ae9951388c1e26241f368c279.filez: Server returned HTTP 404
```

tried ` flatpak update -vy --no-static-deltas org.kde.Platform `
and then again to install but issue is there to... 

Slackware64-current
 QT6.7
plasmashell 5.27.11

**Comment by rizitis** [comment]
Now it build fine. 
1 issue i have is that I cant use localdocs. 
It only accepts /home/$USER as path for localdocs and nothing else...

**Pull request closed without merging**